db/hardware_test.go

```diff
@@ -0,0 +1,240 @@
+// Package db - hardware tests
+// The following tests validate database functionality, using both network and
+// instance payloads. A postgres instance with the tink database schema is
+// required in order to run the following tests.
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"testing"
+
+	_ "github.com/lib/pq"
+)
+
+// TestInsertIntoDB : inserts a single record in the hardware database.
+func TestInsertIntoDB(t *testing.T) {
+	a := newTestParams()
+	defer a.tinkDb.instance.Close()
+
+	if _, err := a.tinkDb.instance.Exec("delete from hardware;"); err != nil {
+		t.Fail()
+	}
+	if err := a.tinkDb.InsertIntoDB(a.ctx.context, a.instanceData); err != nil {
+		t.Logf("%s", a.instanceData)
+		a.logger.Error(err)
+		t.Fail()
+	}
+	if err := a.tinkDb.InsertIntoDB(a.ctx.context, a.hardwareData); err != nil {
+		t.Logf("%s", a.hardwareData)
+		a.logger.Error(err)
+		t.Fail()
+	}
+}
+
```

Comment:
All these `TestGetByMac` tests feel like they can just be done in test matrix and one _real_ func. These tests all have the same common boiler plate.

Reply:
yeah looking at it again I'm more convinced this should really be done as table-driven subtests. Please rework as such.

Reply:
Lets do this in a follow up PR.